### PR TITLE
FQDN checks

### PR DESCRIPTION
**Pull request by GregSutcliffe**
A couple of usability commits - we have so many issues related to a missing FQDN, so I added a test for it. In testing it worked, I noticed the "You're all set!" message is printed even though it bombs out with the fqdn error, so I fixed that too.

### CODE REVIEW

**Comment by domcleal** [comment]
Untested, but :+1: in principle.

**Comment by ekohl** [comment]
:+1: to the principle.

**Comment by GregSutcliffe** [comment]
Would one of you care to test it too? ;)

**Comment by domcleal** [comment]
In a bit!

File: generate_answers.rb [hunk 1]
```diff
@@ -102,6 +106,14 @@ def parting_message
   exit 0
 end
 
+def parting_error
+  say("\n <%= color('There was an error!', :red) %>")
+  say("\n Check the puppet error messages above, and <%= color('#{$outfile}', :cyan) %> for your config.")
+  say("\n Once you have fixed the problem, you can re-run it in the future as root with:")
+  say("  echo include foreman_installer | puppet apply --modulepath #{$modulepath} -v")
+  exit 0
```
Comment (domcleal) [inline generate_answers.rb]: `exit 1`?

File: generate_answers.rb [hunk 2]
```diff
@@ -89,7 +89,11 @@ def save_or_run_and_exit
   $modulepath = File.exists?("#{$workdir}/foreman_installer") ? $workdir : ask("<%= color('Where are the installer Puppet modules?', :question) %>")
   if Process::UID.eid == 0 && agree("\n<%= color('Do you want to run Puppet now with these settings?', :question) %> (y/n)", false)
     system("echo include foreman_installer | puppet apply --modulepath #{$modulepath} -v")
-    parting_message
+    if $?.exitstatus == 0
```
Comment (domcleal) [inline generate_answers.rb]: Perhaps use `--detailed-exitcodes` and then check for either 1, 4 or 6 to indicate failure (or not 0 or 2)?  That way it'll also pick up on errors during the run and not just catalog compilation errors.

**Comment by GregSutcliffe** [comment]
@domcleal sorry for the delay, updated with your suggestions

File: generate_answers.rb [hunk 3]
```diff
@@ -99,6 +103,14 @@ def parting_message
   say("\n Okay, you're all set! Check <%= color('#{$outfile}', :cyan) %> for your config.")
   say("\n You can apply it in the future as root with:")
   say("  echo include foreman_installer | puppet apply --modulepath #{$modulepath} -v")
+  exit 1
```
Comment (domcleal) [inline generate_answers.rb]: exit codes for sucess/error appear to be mixed up

Comment (GregSutcliffe) [inline generate_answers.rb]: Curses, changed the wrong one ;)

**Comment by GregSutcliffe** [comment]
@domcleal oops, fixed.

**Comment by domcleal** [comment]
Thanks, merged!

**Pull request merged**